src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSpark.java

```diff
@@ -140,6 +141,11 @@
             fullName = StandardArgumentDefinitions.METRICS_FILE_LONG_NAME)
     protected String metricsFile;
 
+    @Advanced
+    @Argument(doc = "Override to allow non-queryname sorted inputs for multiple input bams.", optional=true,
+            fullName = "allow-multiple-sort-orders-in-input")
```

Comment:
Lets change this to `treat-unsorted-files-as-queryname-grouped`.  Make it apply to both single and multiple files.  It should only overrule the case where the files are listed as unsorted or don't have a given sort order.  A file that says it's coordinate sorted should still be treated as such.

The comment needs to mention that it will silently give incorrect results if the file's are not actually queryname grouped, and that this is an unsafe workaround for legacy files.

Reply:
It actually doesn't do that. What this change actually does is override the hard failure that used to throw an exception if all the multiple input files didin't have query grouped sort order.

If that override fails, the header associated with the input gets set to "unsorted" with no GO tag, which means the tool treats the input as being non-queryname sorted. Thus it will just happily sort the entire input for you at the outset. What the above check was doing before was silently asserting that the input was GO:query if all the inputs were indeed at least GO:query ordered according to their original headers.